connectivity/builder/builder.go

```diff
@@ -79,23 +79,86 @@ type testBuilder interface {
 	build(ct *check.ConnectivityTest, templates map[string]string)
 }
 
-// NetworkPerformanceTests builds the network performance connectivity tests.
-func NetworkPerformanceTests(ct *check.ConnectivityTest) error {
-	return getTests(ct, []testBuilder{networkPerf{}})
+// GetTestSuites returns a slice of connectivity test builder functions.
+// Each function contains set of tests according to the provided parameters.
+func GetTestSuites(params check.Parameters) ([]func(connTests []*check.ConnectivityTest, extra func(ct *check.ConnectivityTest) error) error, error) {
+	switch {
+	case params.Perf:
+		return []func(connTests []*check.ConnectivityTest, _ func(ct *check.ConnectivityTest) error) error{
+			func(connTests []*check.ConnectivityTest, _ func(ct *check.ConnectivityTest) error) error {
+				return networkPerformanceTests(connTests[0])
+			},
+		}, nil
+	case params.IncludeConnDisruptTest && params.ConnDisruptTestSetup:
+		// Exit early, as --conn-disrupt-test-setup is only needed to deploy pods which
+		// will be used by another invocation of "cli connectivity test"
+		// with include --include-conn-disrupt-test"
+		return []func(connTests []*check.ConnectivityTest, _ func(ct *check.ConnectivityTest) error) error{
+			func(connTests []*check.ConnectivityTest, _ func(ct *check.ConnectivityTest) error) error {
+				return connDisruptTests(connTests[0])
+			},
+		}, nil
+	case params.TestConcurrency > 1:
```

Comment:
This value of `TestConcurrency` is never used to limit the concurrency to a specific value. The only purpose that it is serving with this setup is to run the `concurrentTests` in a separate collection of goroutines with unlimited concurrency, block, and then run the `sequentialTests` with unlimited concurrency.

I would expect this value of `TestConcurrency` to control the number of concurrent tests, which it does not appear to. Unless I'm missing something.

Based on the way that this `builder` is invoked below, even the `sequentialTests` will be invoked with unlimited concurrency by the `runner.MultiError`. Any collection of tests called `sequentialTests` should be run sequentially, with no concurrency at all.

Otherwise it would be better to call these `concurrentSuite1` and `concurrentSuite2`.

I may be missing something so let me know if this is not the behavior that you are seeing.

Reply:
`TestConcurrency` defines the max number of test namespaces that will be provisioned in a cluster.
To run two or > tests can be run concurrently only in separate namespaces to avoid policy collision.

Unfortunately, there are a couple of exceptions that lead to sequential runs:
1. Tests that use cluster-wide network policy
2. Check log test that must be always run at the end